### DISCUSSION TITLE
Faster aircrack-ng

### DISCUSSION THREAD

**Discussion by tehFpa** [aircrack-ng/aircrack-ng#2568] [Ideas]
I made a working version(s) of this method.

Here is a example of what it does:

calc_pmk();
calc_ptk();
decrypt_tkip() or decrypt_ccmp();

wpa3scan.zip // wpa_passphrase essid passphrase & airdecap-ng cap-file.cap -e essid -k pmk
airdecode-ng.zip // using a modified airdecap-ng.c

The code would probably run much faster if, only open cap-file.cap ONCE and try decrypting the packet, instead of reading FULL packet for each pmk!!!

[wpa3scan.zip](https://github.com/aircrack-ng/aircrack-ng/files/12450627/wpa3scan.zip)
[airdecode-ng.zip](https://github.com/aircrack-ng/aircrack-ng/files/12450628/airdecode-ng.zip)

Decoding packet does not require eapol???

**Comment by Mister-X-** [comment]
Have you done any benchmarking to compare aircrack-ng and this? I'd love to see it.

**Reply by tehFpa** [reply]
The project bundled in [wpa3scan.zip](https://github.com/aircrack-ng/aircrack-ng/files/12450627/wpa3scan.zip) is much slower than aircrack-ng

The project bundled in [airdecode-ng.zip](https://github.com/aircrack-ng/aircrack-ng/files/12450628/airdecode-ng.zip) is also slower but much faster than wpa3scan project.

If i could get some clean code, and is minimal resources, that only opens the pcap file once, to compare pmks would be nice.

**Reply by tehFpa** [reply]
I need open pcap file and read necessary data
calc_pmk();
calc_ptk();
decrypt_tkip() and/or decrypt_ccmp();
calc_mic(); maybe?

as well as crypto functions

**Comment by tehFpa** [comment]
[test-calc-mic.c.txt](https://github.com/aircrack-ng/aircrack-ng/files/12734266/test-calc-mic.c.txt)

Trying to merge these files:
[aircrack-ng-1.7/test/cryptounittest/]
calc_pmk
calc_ptk
calc_mic

I am getting data overflows on some variables...
Please help!

**Comment by tehFpa** [comment]
maybe someone can help me finish this project...
![ss](https://github.com/aircrack-ng/aircrack-ng/assets/143378490/4f003c47-857f-4af5-97d1-ea3778f7b0ba)
calc_pmk() & calc_ptk() are working (1 & 2)
calc_mic() computes the wrong MIC (3 & 4)

[test-calc-mic.c.txt](https://github.com/aircrack-ng/aircrack-ng/files/12776569/test-calc-mic.c.txt)
you need to read vars from a file somewhat like 'aircrack-ng -J' no spaces or :'s all on seperate lines

air-j.file:

ESSID
KEYVERSION
BSSID_NO:
STMAC_NO:
SNONCE
ANONCE
KEYMIC
EAPOL

**Comment by tehFpa** [comment]
Finished this project today, it works 'somewhat' fast...
[ aircrack-ng-1.7-win\test\cryptounittest\test-calc-mic.c ]
Usage: ./test-calc-mic air-j.file charset

air-j.file ( example; got values by running 'aircrack-ng capfile-0x.cap -J notused.hccap' , save them to a file format below ):
ESSID
KEYVERSION
BSSID_NO:
STMAC_NO:
SNONCE
ANONCE
KEYMIC
EAPOL

[test-calc-mic.c.txt](https://github.com/aircrack-ng/aircrack-ng/files/12789241/test-calc-mic.c.txt)
